jwst/assign_wcs/tests/test_nirspec.py

```diff
@@ -448,3 +449,114 @@ def test_shutter_size_on_sky():
     assert sep_x.value < 0.194
     assert sep_y.value > 0.45
     assert sep_y.value < 0.46
+
+def test_fixed_slit_functional():
+    #     """
+    #     Compare Nirspec instrument model with IDT model.
+    #     """
+    ins_file = get_file_path('fixed_slits_functional_ESA_v4_20180618.txt')
+    ins_tab = table.Table.read(ins_file, format='ascii')
+
+    hdul = create_nirspec_fs_file(grating='G395H', filter='F290LP')
+    im = datamodels.ImageModel(hdul)
+
+    refs = create_reference_files(im)
+
+    pipeline = nirspec.create_pipeline(im, refs, slit_y_range=[-0.55, 0.55])
+    w = wcs.WCS(pipeline)
+    im.meta.wcs = w
+
+    # Setup the test
+    slitx = [0] * 5
+    slity = [-.5, -.25, 0, .25, .5]
+    lam = np.array([2.9, 3.39, 3.88, 4.37, 5]) * 10**-6
+
+    # Use slit S200A1
+    slit_wcs = nirspec.nrs_wcs_set_input(im, 'S200A1')
+
+    # Slit to MSA absolute
+    slit2msa = slit_wcs.get_transform('slit_frame', 'msa_frame')
+    msax, msay, _ = slit2msa(slitx, slity, lam)
+    assert_allclose(slitx, ins_tab['xslitpos'])
+    assert_allclose(slity, ins_tab['yslitpos'])
+    assert_allclose(msax, ins_tab['xmsapos'])
+    assert_allclose(msay, ins_tab['ymaspos'])
+
+    # Coordinates at Collimator exit
+    # Applies the Collimator forward transform to MSa absolute coordinates
+    col = datamodels.open(refs['collimator'])
+    colx, coly = col.model.inverse(msax, msay)
+    col.close()
```

Comment:
You can use a context manger here:

```
with datamodels.open(refs['collimator']) as col:
    colx, coly = col.model.inverse(msax, msay)
```
and that way if there's a problem opening the reffile, it will get closed on exiting the `with` block.  This is a bit more reliable (and more compact) that explicitly calling `close()`.